### ISSUE TITLE
Occasional request verification failures in a Bolt app

### ISSUE THREAD

**Issue by mfuntowicz** [slackapi/java-slack-sdk#425]
### Issue Type

- [x] Bug
- [ ] Enhancement / Feature request
- [ ] Question
- [ ] Documentation

### Description

I've a very simple Section + Static Select which when calling back the `blockAction `on Java side results in 

```java
[pool-1-thread-4] INFO com.slack.api.bolt.middleware.builtin.RequestVerification - Invalid signature detected - v0=0142715fd794e612fe3eccbb3d2f62441c55b5bffecb4bed83602fc67d9937c6
[pool-1-thread-4] DEBUG io.micronaut.http.server.netty.RoutingInBoundHandler - Encoding emitted response object [{"error":"invalid request"}] using codec: io.micronaut.runtime.http.codec.TextPlainCodec@5bcbdf3c

```

The Interactive Block looks like the following: 

```kotlin
val opts = listOf(
            OptionObject(plainText("Let's go! :rocket:", true), "report_go", null, null),
            OptionObject(plainText("Snooze ... :zzz:", true), "report_snooze", null, null),
            OptionObject(plainText("Not today ... :smirk:", true), "report_discard", null, null)
        )

        return asBlocks(
            section{ section ->
                section.blockId("report_fill_ask")
                    .text(markdownText("Hey ${user.realName.split(" ")[0]}! ")).accessory(staticSelect{ staticSelect ->
                        staticSelect.placeholder(plainText("Select an item"))
                            .options(opts)
                            .actionId("report-action")
                    })
            }
        )

```
I've checked my Signing Secret is correct on both Slack and Java.

---

The issue is reproducible in:
* Module Version: 1.0.3
* JDK Version: 9
* OS Version etc: Windows 10

### The expected result is:
Validation should be ok.

### The actual result is:
[pool-1-thread-4] INFO com.slack.api.bolt.middleware.builtin.RequestVerification - Invalid signature detected 

### Requirements

* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

**Comment by seratch** [comment]
@mfuntowicz 
With the given information, this looks like a misconfiguration issue on your Slack app. Can you verify the following again?

* Verify if **Request URL** in **Interactivity & Shortcuts** > **Interactivity** is valid
* Verify if the Slack app that serves the **Request URL** is using valid **Signing Secret**

If you use the default constructor of `App` class, your app loads signing secret value from env variables (`SLACK_SIGNING_SECRET`). For some reason, your app may load a different value.

**Comment by mfuntowicz** [comment]
Thanks @seratch to follow up on this.

Just checked again what you suggested, and things are looking correct. In fact, before posting this message I took the time to check all these URLs and OAuth permissions.

Another point, my Slack Bot send multiple differents messages, mainly without any Interaction and these are running fine, I receive them and I'm able to answer them back.

Also, I've some commands with the same callback URL than the one for the Interactive messages, and still, I receive them without any issue.

Indeed, the issue arise only for this specific message, so I dug little bit further: 

It appears everything is working fine if the first section contains `plainText` object for `text` attribute. The issue shows up when changing `plainText` to `markDown` and insering a *bold* element inside. 


*Works*

```kotlin
asBlocks(
                section{ section ->
                    section.text(markdownText("Hey ${user.realName.split(" ")[0]}! " +
                            "This is a test message wil bold"
                    ))
                    .accessory(staticSelect{ staticSelect ->
                        staticSelect.placeholder(plainText("Select an item"))
                            .options(opts)
                            .actionId(JARVIS_REPORT_BLOCK_ACTION)
                    })
                }
)
```

*Doesn't work (ie: Invalid Signature)*

```kotlin
asBlocks(
             section{ section ->
                    section.text(markdownText("Hey ${user.realName.split(" ")[0]}! " +
                            "This is a test message with *bold* inside"
                    ))
                    .accessory(staticSelect{ staticSelect ->
                        staticSelect.placeholder(plainText("Select an item"))
                            .options(opts)
                            .actionId("test_action")
                    })
                }
)
```

**Comment by seratch** [comment]
@mfuntowicz Just in case, I verified a tiny micronaut app including your samples works as below.

I don't think the root cause of your issue is related to the blocks. Something is wrong with your app's configuration or your server.

```java
import com.slack.api.bolt.App;
import com.slack.api.bolt.AppConfig;
import com.slack.api.model.block.LayoutBlock;
import com.slack.api.model.block.SectionBlock;
import com.slack.api.model.block.composition.OptionObject;
import io.micronaut.context.annotation.Factory;

import javax.inject.Singleton;
import java.util.Arrays;
import java.util.List;

import static com.slack.api.model.block.Blocks.asBlocks;
import static com.slack.api.model.block.Blocks.section;
import static com.slack.api.model.block.composition.BlockCompositions.*;
import static com.slack.api.model.block.element.BlockElements.staticSelect;

@Factory
public class AppFactory {

  @Singleton
  public AppConfig buildConfig() {
    return AppConfig.builder().build();
  }

  @Singleton
  public App buildApp(AppConfig config) {
    App app = new App(config);
    app.command("/test-test", (req, ctx) -> ctx.ack(buildMessage(req.getPayload().getUserName())));
    app.blockAction("test_action", (req, ctx) -> {
      ctx.respond(r -> r.text("thanks!"));
      return ctx.ack();
    });
    return app;
  }

  private static final List<OptionObject> options = Arrays.asList(
    option(plainText("Let's go! :rocket:", true), "report_go"),
    option(plainText("Snooze ... :zzz:", true), "report_snooze"),
    option(plainText("Not today ... :smirk:", true), "report_discard")
  );

  private static List<LayoutBlock> buildMessage(String name) {
      SectionBlock section = section(s -> s
        .text(markdownText("Hey " + name + "! This is a test message with *bold* inside"))
        .accessory(staticSelect(ss -> ss
          .placeholder(plainText("Select an item"))
          .options(options)
          .actionId("test_action"))
        ));
      return asBlocks(section);
  }
}
```

<img width="500" src="https://user-images.githubusercontent.com/19658/78760993-ef127c00-79bc-11ea-9c77-0be6470cff64.png">

**Comment by mfuntowicz** [comment]
Hum, interesting! Thanks for trying out. 

I'll dig further and come back to you to keep you informed.

Again thanks for the support !

**Comment by seratch** [comment]
@mfuntowicz 
Now I realized that this is caused by the same bug you reported at #426. That makes sense a lot. Somehow I went with the wrong way to implement the adapter for Micronaut and tests were not enough. 

I apologize for the confusion and saying this is not a bug. I will merge #427 soon and release a new patch version today.

**Comment by mfuntowicz** [comment]
Thanks @seratch ! And no worries, happy we were able to track this down for both issues. 

Can you ping me here when the patch is up ? I'll give it a try 👍

**Comment by seratch** [comment]
@mfuntowicz 
I just released a new patch version - 1.0.5 to the Sonatype OSS repository. The jar files will be synced to the Maven Central soon. 

Let me close this issue now but if you see something related to the new version, please feel free to reopen this. Thanks again for your great help on this!
